tests/unit-tests/test_utils.py

```diff
@@ -1251,27 +1252,151 @@ def create_dummy_file(file_path):
+        assert os.path.exists(output_file)
+    
+        tar_object = tarfile.open(output_file)
+        contents = tar_object.getmembers()
+        results = {content.name: content.isfile() for content in contents}
+
+        assert 'code' in results        
+        assert 'code/nc_infer_entry.py' in results and \
+               results['code/nc_infer_entry.py'] == True
+        assert 'model.bin' in results and results['model.bin'] == True
+        assert 'test.yaml' in results and results['test.yaml'] == True
+        assert 'test.json' in results and results['test.json'] == True
+        tar_object.close()
+
+    # test case 2: abnormal cases
+    #     2.1: missing one of the four files
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        os.makedirs(tmpdirname, exist_ok=True)
+        entry_path = os.path.join(tmpdirname, 'nc_infer_entry.py')
+        model_path = os.path.join(tmpdirname, 'model.bin')
+        yaml_path = os.path.join(tmpdirname, 'test.yaml')
+        json_path = os.path.join(tmpdirname, 'test.json')
+
+        output_path = os.path.join(tmpdirname, 'output_folder')
+        os.makedirs(output_path, exist_ok=True)
+
+        # not create the entry point file
+        create_dummy_file(model_path)
+        create_dummy_file(yaml_path)
+        create_dummy_file(json_path)
+        with assert_raises(AssertionError):
+            wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                        output_package_name='model', output_path=output_path)
+
+        os.remove(model_path)
+        create_dummy_file(entry_path)
+        # not create the model.bin file.
+        create_dummy_file(yaml_path)
+        create_dummy_file(json_path)
+        with assert_raises(AssertionError):
+            wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                        output_package_name='model', output_path=output_path)
+
+        os.remove(yaml_path)
+        create_dummy_file(entry_path)
+        create_dummy_file(model_path)
+        # not create the test.yaml file.
+        create_dummy_file(json_path)
+        with assert_raises(AssertionError):
+            wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                        output_package_name='model', output_path=output_path)
+
+        os.remove(json_path)
+        create_dummy_file(entry_path)
+        create_dummy_file(model_path)
+        create_dummy_file(yaml_path)
+        # not create the test.json file.
+        with assert_raises(AssertionError):
+            wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                        output_package_name='model', output_path=output_path)
+
+    #     2.2: missing output folder
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        os.makedirs(tmpdirname, exist_ok=True)
```

Comment:
Is this needed? tempfile will create the temp directory right? Same for other occurrences of this

Reply:
Good catch. Removed all.